### PR TITLE
chapter 4

### PR DESCRIPTION
**Pull request by Rashdan98**

### CODE REVIEW

**Comment by Rashdan98** [comment]
ready

**Review by ghost — COMMENTED** [review]
fix conflict please

**Comment by Rashdan98** [comment]
ready

File: app/helpers/static_pages_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
+end
```
Comment (ngocvt-0484) [inline app/helpers/static_pages_helper.rb]: remove this file because not use

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
-end
+  def full_title page_title = ""
+    base_title = "Ruby on Rails Tutorial Sample App"
+    page_title.empty? ? base_title : "#{page_title} | #{base_title}"
+  end
+end
```
Comment (ngocvt-0484) [inline app/helpers/application_helper.rb]: please add end line
image

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
-
   <body>
-    <%= yield %>
+    <%= render "layouts/header" %>
+    <div class="container">
+      <%= yield %>
+    </div>
   </body>
-</html>
+</html>
```
Comment (ngocvt-0484) [inline app/views/layouts/application.html.erb]: missing end line

File: app/helpers/static_pages_helper.rb (hunk 1 above)
Comment (Rashdan98) [reply app/helpers/static_pages_helper.rb]: noted

File: app/helpers/application_helper.rb (hunk 2 above)
Comment (Rashdan98) [reply app/helpers/application_helper.rb]: alright

File: app/views/layouts/application.html.erb (hunk 3 above)
Comment (Rashdan98) [reply app/views/layouts/application.html.erb]: done

**Comment by Rashdan98** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 4]
```diff
@@ -0,0 +1,37 @@
+/* universal */
+
+body {
+    padding-top: 60px;
```
Comment (huongnt-2545) [inline app/assets/stylesheets/custom.scss]: indent from here to finish, indent = 2 spaces

File: app/assets/stylesheets/custom.scss [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    text-transform: uppercase;
+    letter-spacing: -1px;
+    padding-top: 9px;
+    font-weight: bold;
+  }
+  
+  #logo:hover {
+    color: #fff;
+    text-decoration: none;
+  }
```
Comment (huongnt-2545) [inline app/assets/stylesheets/custom.scss]: add empty line at the end of file

File: app/assets/stylesheets/static_pages.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
```
Comment (huongnt-2545) [inline app/assets/stylesheets/static_pages.scss]: delete this file, it's not useful

File: app/controllers/static_pages_controller.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+class StaticPagesController < ApplicationController
+  def home
+  end
+
+  def help
+  end
+end
```
Comment (huongnt-2545) [inline app/controllers/static_pages_controller.rb]: add empty line at the end of file

File: app/views/layouts/application.html.erb (hunk 3 above)
Comment (huongnt-2545) [reply app/views/layouts/application.html.erb]: still not fix yet, add new empty line at the end of file

File: app/views/static_pages/help.html.erb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+<%= provide :title, "Help"%>
+
+<h1>Help</h1>
+<p>
+  Get help on the Ruby on Rails Tutorial at the
+  <a href="http://www.railstutorial.org/help">Rails Tutorial help page</a>.
+  To get help on this sample app, see the
+  <a href="http://www.railstutorial.org/book"><em>Ruby on Rails
+  Tutorial</em> book</a>.
+</p>
```
Comment (huongnt-2545) [inline app/views/static_pages/help.html.erb]: the same

File: app/views/static_pages/home.html.erb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+<%= provide :title, "Home"%>
+
+<h1>Sample App</h1>
+<p>
+  This is the home page for the
+  <a href="http://www.railstutorial.org/">Ruby on
+  Rails Tutorial</a>
+  sample application.
+</p>
```
Comment (huongnt-2545) [inline app/views/static_pages/home.html.erb]: the same

File: config/routes.rb [hunk 10]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
-end
+  root 'static_pages#home'
+  get 'static_pages/help'
+end
```
Comment (huongnt-2545) [inline config/routes.rb]: the same

File: test/controllers/static_pages_controller_test.rb [hunk 11]
```diff
@@ -0,0 +1,13 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/controllers/static_pages_controller_test.rb]: delete this file, it's not useful

**Comment by Rashdan98** [comment]
ready

**Review by huongnt-2545 — COMMENTED** [review]
This pull request has conflict, you need to resolve it
steps:
```
- git checkout main
- git pull origin main
- git checkout chapter_4
- git rebase main
- # Resolve all conflict files, after that
git add .
git rebase --continue
- git push origin chapter_4 -f
```

**Comment by Rashdan98** [comment]
ready

**Comment by Rashdan98** [comment]
ready

**Comment by Rashdan98** [comment]
ready

**Comment by Rashdan98** [comment]
ready

**Comment by Rashdan98** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 12]
```diff
@@ -0,0 +1,37 @@
+/* universal */
+
+body {
+            padding-top: 60px;
```
Comment (huongnt-2545) [inline app/assets/stylesheets/custom.scss]: wrong indent
fix the same error in this file
```suggestion
  padding-top: 60px;
```

File: app/controllers/static_pages_controller.rb [hunk 13]
```diff
@@ -1,5 +1,8 @@
...
-  def home; end
+  def home
+  end
 
-  def help; end
-end 
+  def help
+  end
+end
+
```
Comment (huongnt-2545) [inline app/controllers/static_pages_controller.rb]: remove all redundant spaces in line at the end of file

**Review by huongnt-2545 — COMMENTED** [review]
This pull have 2 commits, need to combine in 1

**Comment by Rashdan98** [comment]
ready

**Pull request closed without merging**